### PR TITLE
Suggested changes for HPX V1.5 release notes

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: docs/sphinx/releases/whats_new_1_5_0.rst [hunk 1]
```diff
@@ -21,39 +21,48 @@ conformant. As part of the modularization we have begun specifying the public
...
 * We added serialization support for ``std::variant`` and ``std::tuple``.
 * ``result_of`` and ``is_callable`` are now deprecated and replaced by
   ``invoke_result`` and ``is_invocable`` to conform to C++20.
 * We continued with the modularization, making it easier for us to add the new
   experimental ``HPX_WITH_DISTRIBUTED_RUNTIME`` |cmake| option (see below) . An
-  important amount of headers have been deprecated. We adapted the namespaces
+  significant amount of headers have been deprecated. We adapted the namespaces
   and headers we could to be closer to the standard ones (:ref:`api
-  <_public_api>`).
+  <_public_api>`). Depending code should stil compile, however warnings are now
```
Comment (msimberg) [inline docs/sphinx/releases/whats_new_1_5_0.rst:54]:
```suggestion
  <_public_api>`). Depending code should still compile, however warnings are now
```

File: docs/sphinx/releases/whats_new_1_5_0.rst [hunk 2]
```diff
@@ -21,39 +21,48 @@ conformant. As part of the modularization we have begun specifying the public
...
+  generated instructing to change the ``#include``s accordingly.
```
Comment (msimberg) [inline docs/sphinx/releases/whats_new_1_5_0.rst:55]:
```suggestion
  generated instructing to change the includes accordingly.
```

File: docs/sphinx/releases/whats_new_1_5_0.rst [hunk 3]
```diff
@@ -68,33 +77,44 @@ contains official |cmake| targets, performance improvements, and many bug fixes.
...
+* We have added performance counters type information (``aggregating``,
+  ``monotonically increasing``, ``average count``, ``average timer``, etc.).
+* HPX threads are now re-scheduled on the same worker thread they were suspended
   on to avoid cache misses from moving from one thread to the other. This
-  behavior doesn't prevent the task from being stolen.
+  behavior doesn't prevent the thread from being stolen, however.
 * We have added a new configuration option ``hpx.exception_verbosity`` to allow
-  to control the level of verbosity of the exceptions (3 levels available).
+  to control the level of verbosity of the exceptions (3 levels available). The
+  default output generated in case of exceptions is greatly reduced, please use
```
Comment (msimberg) [inline docs/sphinx/releases/whats_new_1_5_0.rst:94]: The default is the the same old verbose output, i.e. unchanged.

File: docs/sphinx/releases/whats_new_1_5_0.rst [hunk 4]
```diff
@@ -107,20 +127,34 @@ contains official |cmake| targets, performance improvements, and many bug fixes.
 * A new example is demonstrating how to disable thread stealing during the
   execution of parallel algorithms
   (``quickstart/disable_thread_stealing_executor.cpp``)
+* We now require for our |cmake| build system configuration files to be
+  formatted using cmake-format.
+* We have removed more dependencies on various Boost libraries. If compiled
+  in C++17 mode (``-DCMAKE_CXX_STANDARD=17``), HPX now does not require linking
+  with any Boost libraries anymore. The library still depends on a number of
+  header-only Boost libraries.
```
Comment (msimberg) [inline docs/sphinx/releases/whats_new_1_5_0.rst:135]: This part about Boost is not new and not accurate. It also still requires a new enough Boost version and a new enough compiler (support for `std::filesystem`).

**Comment by hkaiser** [comment]
I have addressed the review comments, thanks!

**Comment by aurianer** [comment]
Thanks for doing this :)

**Pull request merged**